script.py

```diff
@@ -52,73 +52,117 @@ def check_data_frame(data_frame, spec):
+    return folium_map
+
+def get_format_path_data():
+    trips = get_trips()
+    trips = identify_discrete_trips(trips)
+    trips = filter_trips_by_hour(trips, 12)
+    trips = count_trips(trips)
+    trips = filter_trips_by_count(trips, 5)
+    stations = get_and_format_station_data()
+    trips = format_path_data(trips)
+    trips = add_lat_and_lon_to_path_data_frame(trips, stations)
+    return trips
+
+
+def get_trips():
+    stations = get_station_list()
+    trips = add_lat_and_lon_to_trips(get_and_format_trip_data("resources/Divvy_Trips_2018_06.csv"), stations)
+    return trips
+
+def find_max_and_min_lat_and_lon(trips):
     min_lat = trips["Start_Latitude"].min()
     max_lat = trips["Start_Latitude"].max()
     max_lon = trips["Start_Longitude"].max()
     min_lon = trips["Start_Longitude"].min()
-
-    # make a list of locations (latitude longitude) for each station id
-    locations = trips.groupby("from_station_id").mean()
-
-    locations = locations.loc[:,["Start_Latitude", "Start_Longitude"]]
-
-    trips["path_id"] = [(id1,id2) for id1,id2 in zip(trips["from_station_id"],
-                                                         trips["to_station_id"])]
-
-    paths = trips[trips["hour"]==9].groupby("path_id").count().iloc[:,[1]]
-
-    paths.columns = ["Trip Count"]
-
-    # select only paths with more than X trips
-    paths = paths[paths["Trip Count"]>5]
-    paths["from_station_id"] = paths.index.map(lambda x:x[0])
-    paths["to_station_id"] = paths.index.map(lambda x:x[1])
-
-    paths = paths[paths["from_station_id"]!=paths["to_station_id"]]
-    # # join latitude/longitude into new table
-    paths = paths.join(locations,on="from_station_id")
-
-    locations.columns = ["End Station Latitude","End Station Longitude"]
-    paths = paths.join(locations,on="to_station_id")
-
+    return {min_lat, max_lat, max_lon, min_lon}
+
+def identify_discrete_trips(trips):
+    trips["path_id"] = [(id1, id2) for id1, id2 in zip(trips["from_station_id"],
+                                                       trips["to_station_id"])]
+    return trips
+
+def filter_trips_by_hour(trips, hour):
+    trips = trips[trips["hour"] == hour]
+    return trips
+
+def count_trips(trips):
+    paths = trips.groupby("path_id").count()
+    paths = paths.iloc[:, [1]]
+    paths.columns = ['Trip Count']
+    return paths
+
+def filter_trips_by_count(paths, min_count):
+    paths = paths[paths["Trip Count"] > min_count]
+    return paths
+
+def get_and_format_station_data():
+    locations = get_station_list()
+    locations = locations.loc[:, ['latitude', 'longitude']]
+    locations.columns = ['Start_Latitude', 'Start_Longitude']
+    return locations
+
+def format_path_data(paths):
+    paths["from_station_id"] = paths.index.map(lambda x: x[0])
+    paths["to_station_id"] = paths.index.map(lambda x: x[1])
+    paths = paths[paths["from_station_id"] != paths["to_station_id"]]
```

Comment:
This is filtering the dataframe, which I'd have never guessed from the method name, we'll think more about naming here in general (though it's very hard!)